sdks/python/apache_beam/testing/synthetic_pipeline_test.py

```diff
@@ -57,29 +57,39 @@ class SyntheticPipelineTest(unittest.TestCase):
 
   # pylint: disable=expression-not-assigned
 
-  def test_synthetic_step(self):
-    start = time.time()
+  def test_count_elements_after_synthetic_step(self):
```

Comment:
The best test names describe what the test does without the need for looking into the code. I suggest `test_synthetic_step_yields_10_elements_per_input_record`, as well as `test_synthetic_sdf_step_yields_10_elements_per_input_record` in the similar test below. WDYT?

Reply:
Fair point. Well, to be honest my `description what test does without looking into the code` was more about the state that is tested/expected so if the count is correct. But this is just different approach. We would be more precise and say `test_synthetic_step_returns_10_elements_per_10_input_record` what actually explains what is deeper in `SyntheticStep` implementation or just assume that we expect 100 records at the end and if someone wants more details will read the test: `test_synthetic_step_returns_100_elements`.

Reply:
Imagine that you have to test a completely different behaviour. You add a new test which covers it and asserts that the number of output elements is equal to 20 (or anything else). Then the test `test_synthetic_step_returns_20_elements` would be not distinguishable from the `test_synthetic_step_returns_100_elements` without investigating the code. And we want to avoid this.

Reply:
To be honest I am not really sure if the numbers here are essential, but more if the count is just correct (that's why I suggested the original name). I can agree, if there will be similar test that will also return 100 elements, it would have to be distinguished somehow. So what do you think about compromise where we focus what is tested and we skip numbers, because those aren't important in this case? By the rule `test[Feature being tested]` we can have something like: `test_synthetic_step_multiplies_output_elements_count`.

Reply:
Allright. Looks good for me